pallets/did/src/lib.rs

```diff
@@ -1254,11 +1268,62 @@ pub mod pallet {
 
 			Ok(())
 		}
+
+		#[pallet::call_index(17)]
+		// TODO: Benchmark base call dispatch + the weight specified in the call
+		#[pallet::weight(*max_weight)]
+		pub fn cleanup_linked_resources(
+			origin: OriginFor<T>,
+			did: DidIdentifierOf<T>,
+			max_weight: Weight,
+		) -> DispatchResult {
+			// If it's a regular transaction, verify the submitter is the deposit owner.
+			if let Ok(sender) = ensure_signed(origin.clone()) {
+				let did_entry = Did::<T>::get(&did).ok_or(Error::<T>::NotFound)?;
+				ensure!(did_entry.deposit.owner == sender, Error::<T>::BadDidOrigin);
+			// If it's a DID-authenticated operation, verify the specified `did`
+			// matches the origin.
+			} else if let Ok(did_origin) = T::EnsureOrigin::ensure_origin(origin) {
+				ensure!(did_origin.subject() == did, Error::<T>::BadDidOrigin);
+			// We don't accept any other origin -> fail.
+			} else {
+				return Err(DispatchError::BadOrigin);
+			};
+
+			// The cleanup logic does not need to get benchmarked as it consumes the
+			// provided weight.
```

Comment:
I think this is not necessary true. `remaining_weight` is only getting updated by the performed operation or not? 

So the processing in the loop and the event creation has to be benchmarked or not?

Reply:
Yes this would need to be refined properly, this is just a pre-draft, but I wanted to start a conversation about it.